_pytest/main.py

```diff
@@ -90,10 +90,12 @@ def wrap_session(config, doit):
             session.exitstatus = doit(config, session) or 0
         except pytest.UsageError:
             raise
-        except KeyboardInterrupt:
+        except KeyboardInterrupt as e:
             excinfo = _pytest._code.ExceptionInfo()
```

Comment:
Yes thanks for bringing this up again, I think it may have to do with the init_state logic that we were talking about prior to your departure.

I will do more work on this for sure.